cli/tests/pcluster/schemas/test_cluster_schema.py

```diff
@@ -316,6 +318,104 @@ def test_scheduling_schema(mocker, config_dict, failure_message):
         SchedulingSchema().load(config_dict)
 
 
+@pytest.mark.parametrize(
+    "config_dict, failure_message",
+    [
+        # Single Instance Type in a Compute Resource
+        (
+            {
+                "Name": "Standard-Queue",
+                "Networking": {"SubnetIds": ["subnet-12345678"]},
+                "ComputeResources": [
+                    {"Name": "compute_resource1", "InstanceType": "c5.2xlarge", "MaxCount": 5},
+                    {"Name": "compute_resource2", "InstanceType": "c4.2xlarge"},
+                ],
+            },
+            "",
+        ),
+        # Multiple Instance Types in a Compute Resource
+        (
+            {
+                "Name": "Flex-Queue",
+                "Networking": {"SubnetIds": ["subnet-12345678"]},
+                "AllocationStrategy": "lowest-price",
+                "ComputeResources": [
+                    {
+                        "Name": "compute_resource1",
+                        "InstanceTypeList": [{"InstanceType": "c5.2xlarge"}],
+                    }
+                ],
+            },
+            "",
+        ),
+        # Mixing Single plus Flexible Instance Type Compute Resources
+        (
+            {
+                "Name": "Flex-Queue",
+                "Networking": {"SubnetIds": ["subnet-12345678"]},
+                "AllocationStrategy": "lowest-price",
+                "ComputeResources": [
+                    {
+                        "Name": "compute_resource1",
+                        "InstanceTypeList": [{"InstanceType": "c5.2xlarge"}, {"InstanceType": "c4.2xlarge"}],
+                    },
+                    {"Name": "compute_resource2", "InstanceType": "c4.2xlarge"},
+                ],
+            },
+            "",
+        ),
+        # Failing to specify either InstanceType or InstanceTypes should return a validation error
```

Comment:
add test case where both (InstanceType or InstanceTypeList) are specified (for the same compute resource), this should return a validation error

Reply:
also fix comment InstanceTypes -> InstanceType or InstanceTypeList